### DISCUSSION TITLE
LaTex objects should have their own color and default color

### DISCUSSION THREAD

**Discussion by kacago22** [xournalpp/xournalpp#6091] [Ideas]
LaTex objects take their default color from the default color of text objects. If one modifies the default color for texts,  when tries do edit a LaTex object then the object changes color to whatever the default text object color is at the moment. It is also difficult to edit the color of a LaTex object (when one selects one the toolbar with colors is disabled). 

Thanks for considering this idea!

**Comment by rolandlo** [comment]
I agree. It shouldn't be hard to implement.

**Comment by MiltonBalaOfficial** [comment]
> LaTex objects take their default color from the default color of text objects. If one modifies the default color for texts, when tries do edit a LaTex object then the object changes color to whatever the default text object color is at the moment. It is also difficult to edit the color of a LaTex object (when one selects one the toolbar with colors is disabled).
> 
> Thanks for considering this idea!

I also feel the need of the feature. This plugin [LaTexWithCurrentColor](https://github.com/MiltonBalaOfficial/LaTexWithCurrentColor) may help a little bit..

**Comment by tattsan** [comment]
You can set your default color in your latex template file like this:
```
--- default_template.tex        2024-11-27 01:18:23.000000000 +0900
+++ your_template.tex   2024-11-29 07:02:25.439909330 +0900
@@ -20,7 +20,34 @@

 % Color support
 \usepackage{xcolor}
-\definecolor{xpp_font_color}{HTML}{%%XPP_TEXT_COLOR%%}
+\newcommand*{\XppTextColor}{%%XPP_TEXT_COLOR%%}
+\newcommand*{\XppLatexColorDefault}{002e99}%%%% Dark Blue in default palette.
+\definecolor{xpp_font_color}{HTML}{\XppLatexColorDefault}
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\definecolor{xyellow}{HTML}{ffe16b}%% Yellow in default palette.
+\definecolor{xorange}{HTML}{ffa154}%% Orange in default palette.
+\definecolor{xmagenta}{HTML}{cd9ef7}% Purple in default palette.
+\definecolor{xlime}{HTML}{9bdb4d}%%%% Light Green in default palette.
+\definecolor{xcyan}{HTML}{64baff}%%%% Light Blue in default palette.
+\definecolor{xgray}{HTML}{808080}%%%% Gray in default palette.
+\definecolor{xgreen}{HTML}{3a9104}%%% Dark Green in default palette.
+\definecolor{xred}{HTML}{ed5353}%%%%% Red in default palette.
+\definecolor{xblue}{HTML}{002e99}%%%% Dark Blue in default palette.
+\definecolor{xblack}{HTML}{000000}%%% Black in default palette.
+\definecolor{xwhite}{HTML}{ffffff}%%% White in default palette.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\providecommand*{\xyellow}{\color{xyellow}}
+\providecommand*{\xorange}{\color{xorange}}
+\providecommand*{\xmagenta}{\color{xmagenta}}
+\providecommand*{\xlime}{\color{xlime}}
+\providecommand*{\xcyan}{\color{xcyan}}
+\providecommand*{\xgray}{\color{xgray}}
+\providecommand*{\xgreen}{\color{xgreen}}
+\providecommand*{\xred}{\color{xred}}
+\providecommand*{\xblue}{\color{xblue}}
+\providecommand*{\xblack}{\color{xblack}}
+\providecommand*{\xwhite}{\color{xwhite}}
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

 % User input
 \begin{scontents}[store-env=preview]
```
And you can change colors in your LaTeX source file with the latex commands like `\xcyan` etc. The disadvantage of this method is that it cannot be synchronized with palette selection; it would be nice to be able to refer to the color number in the palette, like the color reference in toolbar.ini.

**Reply by MiltonBalaOfficial** [reply]
how can I set a increased font size in the template file?

**Reply by tattsan** [reply]
It is discussed on #3858.
